### ISSUE TITLE
how to add jar dependency with npm cli

### ISSUE THREAD

**Issue by jiyinyiyong** [thheller/shadow-cljs#12]
Based on https://github.com/thheller/shadow-cljs/wiki/ClojureScript-for-JS-Devs

How do I add dependencies I released on Clojars?

**Comment by thheller** [comment]
Add it to your `"shadow-cljs"` `"dependencies"` in your `package.json`, basically a jsony version of a `leiningen` artifact.

`[artifact/name "1.0.0"]` becomes `["artifact/name", "1.0.0"]`.

Example:
```
  "shadow-cljs": {
    "dependencies": ["artifact/name", "1.0.0"]
  }
```

**Comment by thheller** [comment]
Oops, there is an array missing.

```
  "shadow-cljs": {
    "dependencies": [
      ["artifact/name", "1.0.0"]
    ]
  }
```

**Comment by jiyinyiyong** [comment]
Tried compiling with my own project https://github.com/mvc-works/stack-workflow/tree/shadow-cljs and got no error... surprise! I will try MHR in Webpack later to see how is the develop experience.

**Comment by jiyinyiyong** [comment]
Tried but HMR still does not work correctly. I think my configs for Webpack has problems(I can't suppose Webpack has bugs).

The code is here:

https://github.com/mvc-works/stack-workflow/tree/shadow-cljs

```bash
yarn
shadow-cljs --once
env=dev node node_modules/shadow-cljs/client.render.js
./node_modules/.bin/webpack-dev-server --config webpack.dev.coffee
open http://localhost:8080/dev.html
```

At here, the HMR looks fine. But it does not work as I edit the file:

```
node_modules/shadow-cljs/client.comp.container.js
```

for example this expression in the file above:

```js
respo.comp.text.comp_text.call(null,"Demo",null)
```

The logs look like this as I change `"Demo"` to something else:

```
2main.js:60631 [WDS] App updated. Recompiling...
main.js:60631 [WDS] App hot update...
main.js:61075 [HMR] Checking for updates on the server...
main.js:4611 Code updated.
main.js:61118 [HMR] Updated modules:
main.js:61120 [HMR]  - ./node_modules/shadow-cljs/client.comp.container.js
main.js:61057 [HMR] App is up to date.
2main.js:60631 [WDS] App updated. Recompiling...
main.js:60631 [WDS] App hot update...
main.js:61075 [HMR] Checking for updates on the server...
main.js:61118 [HMR] Updated modules:
main.js:61120 [HMR]  - ./node_modules/shadow-cljs/client.comp.container.js
main.js:61057 [HMR] App is up to date.
2main.js:60631 [WDS] App updated. Recompiling...
main.js:60631 [WDS] App hot update...
main.js:61075 [HMR] Checking for updates on the server...
main.js:61118 [HMR] Updated modules:
main.js:61120 [HMR]  - ./node_modules/shadow-cljs/client.comp.container.js
main.js:61057 [HMR] App is up to date.
```

However `Code updated` appeared for only once, my DOM does not change. That's unexpected.

Still I think there must be a problem in Webpack configs... need time to see what's happening.

---

As I tried `shadow-cljs --dev`, I found when I edited 'client.comp.container.cljs', both `client.comp.container.js` and `client.main.js` was changed. That correct?

**Comment by thheller** [comment]
Yeah, I have no idea how to do the webpack HMR stuff correctly. Tried a few things but got frustrated very quickly.

Instead I rewrote a few things and added the suggested `shadow-cljs` workflow. Since you are already CLJS only the `:browser` target is perfect for this.

It still uses `webpack` with HMR for the styles stuff ... I think. There are no styles so I didn't test it.

Continue here: https://github.com/mvc-works/stack-workflow/pull/2
